authLdap.php

```diff
@@ -356,6 +363,67 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
 }
 
 /**
+ * Get user's user id
+ *
+ * Returns null if username not found
+ *
+ * @param string $username username
+ * @param string user id, null if not found
+ */
+function authLdap_get_uid($username) {
+    global $wpdb;
+    static $cache = array();
+
+    if (!isset($cache[$username])) {
```

Comment:
Why do you want to cache that information? As far as I see it this is only called once every request. And as the cache is recreated on every request it's more like additional overhead. Or am I missing something?